core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -58,24 +60,26 @@ class TaskInfo(
 
   var failed = false
 
+  var killed = false
+
   private[spark] def markGettingResult(time: Long = System.currentTimeMillis) {
     gettingResultTime = time
   }
 
-  private[spark] def markSuccessful(time: Long = System.currentTimeMillis) {
+  private[spark] def markFinished(time: Long = System.currentTimeMillis, state: TaskState) {
```

Comment:
Can you move state to be first? We usually put default arguments last, and then you don't need to always name the "state" argument when calling this method.